### DISCUSSION TITLE
Is it a big deal to rename a plugin?

### DISCUSSION THREAD

**Discussion by jeffwright13** [pytest-dev/pytest#9841] [General] [answered]
I've been working on a plugin for a while now and the more I get into it, the more I realize that the original name I picked out isn't really indicative of its true functionality now. How big a deal is to just rename the repo in Githib? Will the Pytest auto-generated plugin list be updated (retiring the old name and putting in the new one)? I don't have much of a userbase yet (if any at all).

**Comment by nicoddemus** [comment, marked as answer]
The plugin list is obtained from PyPI actually. You can't rename PyPI packages (AFAIK), so if you already published the plugin, it will appear twice in the plugin list as there will be now to PyPI packages.

**Reply by jeffwright13** [reply]
I see. Well, maybe I can't rename a package but I could probably delete it. In that case, is the list updated to remove 'dead' packages?

**Reply by nicoddemus** [reply]
Don't know for sure, but that's easy to update if not. 👍

**Reply by The-Compiler** [reply]
Our plugin list script already excludes plugins with a development status of `Development Status :: 7 - Inactive`. Thus, making a final release with the old name and that classifier set would probably be the best way to do this. See the [pypi-rename cookiecutter template](https://github.com/simonw/pypi-rename) for some help with creating such a final release.

**Reply by jeffwright13** [reply]
Ahh, good tip - thanks.
